### PR TITLE
Don't write more data than will fit in flash

### PR DESCRIPTION
**Pull request by lurch**

### CODE REVIEW

File: main.cpp [hunk 1]
```diff
@@ -1720,13 +1720,33 @@ void load_command::execute(device_map &devices) {
...
+        }
+    }
+    if (uses_flash) {
+        int32_t size_guess = guess_flash_size(raw_access);
+        if (size_guess > 0) {
+            assert(flash_max > flash_min);
+            uint32_t flash_data_size = flash_max - flash_min;
+            assert(flash_min >= FLASH_START);
+            uint32_t flash_start_offset = flash_min - FLASH_START;
+            if ((flash_start_offset + flash_data_size) > size_guess) {
```
Comment (kilograham) [inline main.cpp:1746]: seems like we could just track flash_max (flash_min provides no actual value)

Comment (lurch) [reply main.cpp:1746]: I assumed that the `assert(flash_min >= FLASH_START);` check might be useful? Or should I take that out too?

**Comment by kilograham** [comment]
closing this - as the code s pretty stale... the verify option is a good workaround for this anyway, but feel free to re-open an issue

**Comment by kilograham** [comment]
ha i just got bitten by this myself :-)

**Comment by lurch** [comment]
I've just rebased this onto the latest `develop`.

**Comment by lurch** [comment]
Should I re-open this against `master`, or just leave it closed?

**Comment by kilograham** [comment]
well reopen against develop

**Comment by lurch** [comment]
Ahhh, I saw the "kilograham deleted the branch `raspberrypi:develop` yesterday" message (and I assume that is what auto-closed this PR?), but I didn't realise that you'd actually created a new `develop` branch.

I don't have the permissions to re-open **this** PR, so I guess you could either do that for me, or I could create a fresh PR? (which would mean we'd lose your review-comments) - up to you.

**Comment by lurch** [comment]
BTW it's just occurred to me that if you made `develop` a protected branch, then GitHub probably wouldn't try auto-deleting it each time you merge `develop` into `master`? :slightly_smiling_face:  (and therefore any other PRs against `develop` hopefully won't get auto-closed)

**Pull request closed without merging**